src/games/Lexible/models/PresenterModel.ts

```diff
@@ -690,6 +687,19 @@ export class LexiblePresenterModel extends ClusterfunPresenterModel<LexiblePlaye
         }
     }
 
+    //--------------------------------------------------------------------------------------
+    // 
+    //--------------------------------------------------------------------------------------
+    handleSwitchTeam(player: LexiblePlayer) {
```

Comment:
If the switch team message includes the desired team, then this function should change to either accept or deny the switch request based on the number of players on that team (returning the team that the requesting player ends up on, as we're doing here with the play request message).

(Either that, or accept the switch request, and grey out the "Click here to start!" button with an appropriate error message if all players are on one team. This would allow two lone players to switch teams to match their color preferences).

Also, this action should be denied if the game has already started.

Reply:
I see this as a nice-to-have because the server is the final arbiter of team membership.  We would have to build up and synchronize extra state on the client to make this work properly.  I'd like to leave it as-is for now.

Reply:
Apologies, I just meant to convey that if we decide to implement the changes above to have the "switch teams" message include the desired team, this handler on the presenter should be changed to handle that. Most of this method would end up being the same in that case - the only additional logic would be that if the player is already on the team they want, or if the game has already started, we don't do anything. (In other words, instead of literally sending "accept/deny", the presenter just sends the client what team they're on)

As for the client, we would just calculate what the desired team should be when we send the message - since the client already knows what team they're on, they specify the other one.